scripts/linters/custom_eslint_checks/rules/nested-awaits.js

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Rule to have nested awaits for
+ * await (await browser.switchTo().activeElement()).sendKeys(explanation);
+ * with no exceptions.
+ */
+
+'use strict';
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: `Use nested awaits without any exception for await
+        (await browser.switchTo().activeElement()).sendKeys(explanation);`,
+      category: 'Best Practices',
+      recommended: true,
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      nestedAwaits: `Only use nested awaits for await
+        (await browser.switchTo().activeElement()).sendKeys(explanation);`,
+    },
+  },
+
+  create: function(context) {
+    var activeElementSelector1 = (
+      'CallExpression[callee.property.name = activeElement]' +
+      '[callee.object.callee.object.name = browser]' +
+      '[callee.object.callee.property.name = switchTo]'
+    );
+
+    var activeElementSelector2 = (
+      'CallExpression[callee.object.type = AwaitExpression]' +
+      '[callee.object.argument.callee.property.name = activeElement]' +
+      '[callee.object.argument.callee.object.callee.property.name = switchTo]' +
+      '[callee.object.argument.callee.object.callee.object.name = browser]'
+    );
```

Comment:
Why do we need two selectors? Also, can you please add a clear and meaningful name for variables e.g, `activeElementWIthAwaitSelector`.

Reply:
Why do we need two different selectors for the inner and outer `Await`? (Sorry, I'm finding it difficult to understand, can you please explain this with examples?)

Reply:
@DubeySandeep, That's alright
So we have four cases for this
```
await (await browser.switchTo().activeElement())
await (browser.switchTo().activeElement())
(await browser.switchTo().activeElement())
(browser.switchTo().activeElement())
```
We want the first one.
In second, the inner `Await `is missing. So in this case the inner check gives the error.
In third, the outer `Await `is missing. So in this case the outer check gives the error.
in fourth, both are missing so inner would give an error because it is written first.

If we are trying to check for outer Await, we have two cases:
1. `(browser.switchTo().activeElement())`
2. `(await browser.switchTo().activeElement())`

because of this, we need to have two checks.

Reply:
Just adding a check to ensure `browser.switchTo().activeElement()` call happens with `await` will be sufficient, right? OR am I missing anything?

Reply:
To clear up some of the confusion, I think it's helpful to look at the full code example from the issue:

```
await (await browser.switchTo().activeElement()).sendKeys(explanation);
```

We want to make sure that both `await`s are present. I think what @DubeySandeep is getting at is that we already have other lint checks to make sure that `.sendKeys()` has an await, so we really only need to check that the inner await is present